lib/backend.js

```diff
@@ -450,11 +450,30 @@ exports.createServer = (dynamodb, docClient) => {
           if (indexBeingUsed && indexBeingUsed.KeySchema.find(
             (keySchemaItem) => keySchemaItem.AttributeName === key)
           ) {
-            KeyConditionExpression.push(`#key${i} ${filters[key].operator} :key${i}`)
+            const matchedKeySchema = indexBeingUsed.KeySchema.find(
+              (keySchemaItem) => keySchemaItem.AttributeName === key)
+
+            if (matchedKeySchema.KeyType === 'HASH') {
+              KeyConditionExpression.push(
+                `#key${i} ${filters[key].operator} :key${i}`)
+            } else if (matchedKeySchema.KeyType === 'RANGE') {
+              if (filters[key].operator === 'begins_with') {
+                KeyConditionExpression.push(
+                  `${filters[key].operator} ( #key${i} , :key${i})`)
+              } else {
+                KeyConditionExpression.push(
+                  `#key${i} ${filters[key].operator} :key${i}`)
+              }
+            }
```

Comment:
The code looks very repetitive. Can you clean this up a bit?

The `matchedKeySchema` variable can be initialized before the `if` block.
The `KeyConditionExpression.push` call can be done once after if/else checks if any filter is defined.